packages/frontend/core/src/modules/at-menu-config/services/index.ts

```diff
@@ -369,55 +369,48 @@ export class AtMenuConfigService extends Service {
 
           close();
 
-          if (id === currentUserId) {
-            const inlineRange = inlineEditor.getInlineRange();
-            if (inlineRange && inlineRange.length === 0) {
-              inlineEditor.insertText(inlineRange, ' ', {
-                mention: {
-                  member: id,
-                },
-              });
-              inlineEditor.setInlineRange({
-                index: inlineRange.index + 1,
-                length: 0,
-              });
-            }
+          const inlineRange = inlineEditor.getInlineRange();
+          if (!inlineRange || inlineRange.length !== 0) return;
 
-            return;
-          }
+          inlineEditor.insertText(inlineRange, ' ', {
+            mention: {
+              member: id,
+            },
+          });
+          inlineEditor.setInlineRange({
+            index: inlineRange.index + 1,
+            length: 0,
+          });
 
-          notificationService
-            .mentionUser(id, workspaceId, {
-              id: docId,
-              title: this.docDisplayMetaService.title$(docId).value,
-              blockId: block.blockId,
-              mode: mode as GraphqlDocMode,
-            })
-            .then(notificationId => {
-              const inlineRange = inlineEditor.getInlineRange();
-              if (inlineRange && inlineRange.length === 0) {
-                inlineEditor.insertText(inlineRange, ' ', {
-                  mention: {
-                    member: id,
-                    notification: notificationId,
+          if (id !== currentUserId) {
+            notificationService
+              .mentionUser(id, workspaceId, {
+                id: docId,
+                title: this.docDisplayMetaService.title$(docId).value,
+                blockId: block.blockId,
+                mode: mode as GraphqlDocMode,
+              })
+              .then(notificationId => {
+                inlineEditor.formatText(
```

Comment:
What if the user delete the mention note or inserts text before it during the request?